vpr/src/route/rr_graph_fpga_interchange.cpp

```diff
@@ -0,0 +1,1179 @@
+#include <kj/std/iostream.h>
+#include <zlib.h>
+#include <algorithm>
+#include <map>
+#include <queue>
+
+#include "rr_graph_fpga_interchange.h"
+
+#include "arch_util.h"
+#include "vtr_time.h"
+#include "vtr_error.h"
+#include "globals.h"
+#include "check_rr_graph.h"
+#include "rr_metadata.h"
+#include "rr_graph_indexed_data.h"
+#include "fpga_interchange_arch_utils.h"
+
+using namespace DeviceResources;
+using namespace LogicalNetlist;
+using namespace capnp;
+
+typedef std::tuple<int, int> location;
```

Comment:
Having a `std::pair` for types that only contain two simple types is better as it simplify the access and, instead of

```
std::get<0>(loc)
```

you can simply have:

```
loc.first
```

This would improve readability throughout the code.